client/allocrunner/taskrunner/task_runner.go

```diff
@@ -394,6 +394,17 @@ func (tr *TaskRunner) Run() {
 	defer close(tr.waitCh)
 	var result *drivers.ExitResult
 
+	tr.stateLock.RLock()
+	dead := tr.state.State == structs.TaskStateDead
+	tr.stateLock.RUnlock()
+
+	if dead {
```

Comment:
here I only check if the task itself is dead - I suspect we should be checking if the restore alloc had a terminated alloc state.  I suspect that an alloc with tasks with mixed status causes some some complications?

Reply:
Actually, this is the right behavior.  An alloc is considering running if one task completes, and all allocs will be killed if leader task dies or a task failed enough times.  Until that happens, we should treat other tasks as running.